### PR TITLE
Initial Jest tests for Search Component

### PR DESCRIPTION
**Pull request by joedarc**
Should satisfy #121

### CODE REVIEW

File: client/src/setupTests.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    clear: function() {
+      store = {};
+    }
+  };
+};
+
+global.localStorage = mockLocalStorage();
+global.render = render;
+global.fireEvent = fireEvent;
+global.waitForElement = waitForElement;
```
Comment (peterbe) [inline client/src/setupTests.js]: I don't understand these last 3 assignments to `global`. Unless there's a strong reason for it, I think these should not become global and, instead, in `search.test.js` it should explicitly import the things it needs from `@testing-library/react`.

**Comment by peterbe** [comment]
By the way, this contribution is hugely important because it sets the "template" for how components should be done. Apart from the magic of the `render()` function (from `@testing-library/react`) the rest of the test should ideally be done in a way that we can, some day, replace React with Xeact or Wue or Swelte. 

One thing, I'm interested in contributing on myself is custom sort of search results. Today, when you do a full-text search, it returns the matches based on the full-text score which is an obtuse and almost cryptic formula implemented deep in FlexSearch. What I'd like to do is sort by popularity instead when the results are few. For example just typing "c" should turn up `/en-US/docs/Web/CSS/Specificity` first because it's our most popular page. They probably didn't mean to find `/en-US/docs/Mozilla/js-ctypes` which is what you get today; https://developer.mozilla.org/en-US/search?q=c&topic=apps&topic=html&topic=css&topic=js&topic=api&topic=canvas&topic=svg&topic=webgl&topic=mobile&topic=webdev&topic=http&topic=webext&topic=standards

**Comment by peterbe** [comment]
Looking at the conflict in `client/yarn.lock` it seems the only difference is the babel version. But to be on the safe side I recommend not trying to do it manually. 

1. Manually edit `client/package.json` and remove the new devDependency `"@testing-library/react"`. 
2. Restore `client/yarn.lock` like this: `git checkout origin/master -- client/yarn.lock` (ASSUMING the remote `origin` is the `mdn` repo and not your personal fork)
3. Now rebase. 
4. Now manually put it back in again `cd client && yarn add --dev "@testing-library/react"`

**Comment by peterbe** [comment]
Apparently your first commit isn't signed. But your second one is. You can fix this now by manually squashing these into 1 commit. Then I'll be able to merge. 
Screen Shot 2019-09-23 at 11 38 49 AM

**Comment by joedarc** [comment]
All better 👍

**Pull request merged**